app/assets/javascripts/govuk_publishing_components/analytics/explicit-cross-domain-links.js

```diff
@@ -0,0 +1,43 @@
+/* global GOVUK, $, ga */
+
+(function () {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  var ExplicitCrossDomainLinks = function () {
+    this.decorateForm = function (element, trackers) {
+      if (trackers.length) {
+        var existing = element.attr('action')
+        var linker = new window.gaplugins.Linker(trackers[0])
+        var tracked = linker.decorate(existing)
+        element.attr('action', tracked)
+      }
+    }
+
+    this.decorateLink = function (element, trackers) {
+      if (trackers.length) {
+        var existing = element.attr('href')
+        var linker = new window.gaplugins.Linker(trackers[0])
+        var tracked = linker.decorate(existing)
+        element.attr('href', tracked)
+      }
+    }
+  }
+
+  ExplicitCrossDomainLinks.start = function () {
```

Comment:
If this isn't a GOVUK module, how is this code initialised? I see that it gets included in `analytics.js` but what calls the `start` function to make it run?

Reply:
The answer to this may impact the structure of the test file, as I see you're explicitly calling `decorateLink` (etc.) on the code snippets.

Reply:
Right, so I don't actually know how the module system works, so I put this together based on the ecommerce code, but probably did that wrong.

Reply:
Is there some documentation on how to write a new module?

Reply:
Aha: https://github.com/alphagov/govuk_publishing_components/blob/master/docs/javascript-modules.md

I'll have a read of that and rewrite the code